sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -152,35 +188,34 @@ export class CryptographyClient {
       | [EncryptParameters, EncryptOptions?]
       | [EncryptionAlgorithm, Uint8Array, EncryptOptions?]
   ): Promise<EncryptResult> {
+    this.ensureValid(await this.fetchKey(), KnownKeyOperations.Encrypt);
     const [parameters, options] = this.disambiguateEncryptArguments(args);
 
-    if (this.concreteClient.kind === "remote") {
-      return this.concreteClient.client.encrypt(parameters, options);
-    } else {
-      if (!isLocallySupported(parameters.algorithm)) {
-        throw new Error(
-          `Algorithm ${parameters.algorithm} is not supported for a local JsonWebKey.`
-        );
-      }
-      return this.concreteClient.client.encrypt(parameters, options);
+    const { span, updatedOptions } = this.createSpan("encrypt", options);
+
+    try {
+      const provider = await this.getProvider(KnownKeyOperations.Encrypt, parameters.algorithm);
```

Comment:
would it make sense to cache the provider?

Reply:
That’s a good observation - in this case it wouldn’t because you always want a provider that is appropriate for the algorithm which can be different I think - not today but soon with AES (AES-GCM vs AES-CBC)

but the good news is getProvider caches the provider list so it’s only expensive on the first call and even then only if you passed a key ID (we need to exchange it for the actual key). Everything else doesn’t involve a network call and will reuse cached providers.